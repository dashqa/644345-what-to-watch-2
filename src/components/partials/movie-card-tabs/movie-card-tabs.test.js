import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardTabs from './movie-card-tabs';
import {MovieTabs} from "@constants";

const props = {
  onChangeTab: () => {},
  movie: {
    genre: `Drama`,
    description: `This is very interesting description`,
    released: 1990,
    rating: 6.7,
    runTime: 190,
    director: `Lynne Ramsay`,
    starring: [`Tilda Swinton`, `Ezra Miller`],
    scoresCount: 189999,
  },
  comments: []
};

it(`Movie card tabs correctly renders with overview tab`, () => {
  const active = MovieTabs.OVERVIEW;
  const tree = renderer.create(<MovieCardTabs active={active} {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Movie card tabs correctly renders with details tab`, () => {
  const active = MovieTabs.DETAILS;
  const tree = renderer.create(<MovieCardTabs active={active} {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Movie card tabs correctly renders with review tab`, () => {
  const active = MovieTabs.REVIEWS;
  const tree = renderer.create(<MovieCardTabs active={active} {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
