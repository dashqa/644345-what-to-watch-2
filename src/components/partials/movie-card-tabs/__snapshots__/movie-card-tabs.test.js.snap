// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card tabs correctly renders with details tab 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        189999
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      This is very interesting description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Lynne Ramsay
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tilda Swinton, Ezra Miller
      </strong>
    </p>
  </div>
</div>
`;

exports[`Movie card tabs correctly renders with overview tab 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        189999
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      This is very interesting description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Lynne Ramsay
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tilda Swinton, Ezra Miller
      </strong>
    </p>
  </div>
</div>
`;

exports[`Movie card tabs correctly renders with review tab 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        189999
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      This is very interesting description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Lynne Ramsay
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tilda Swinton, Ezra Miller
      </strong>
    </p>
  </div>
</div>
`;
